packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -1077,6 +1077,11 @@ class TextSelectionGestureDetectorBuilder {
   ///    this callback.
   @protected
   void onDragSelectionStart(DragStartDetails details) {
+    final PointerDeviceKind? kind = details.kind;
+    _shouldShowSelectionToolbar = kind == null
+      || kind == PointerDeviceKind.touch
+      || kind == PointerDeviceKind.stylus;
+
```

Comment:
This code comes from `onTapDown`

Reply:
Issue: https://github.com/flutter/flutter/issues/69127